src/components/layouts/MainHeader.tsx

```diff
@@ -0,0 +1,147 @@
+import { Image, StyleSheet, Text, TouchableOpacity, View } from "react-native";
+import React from "react";
+import DroidconKeIcon from "../../assets/icons/DroidconKeIcon";
+import { layoutProperties } from "../../constants/Properties";
+import { fonts } from "../../assets/fonts/fonts";
+import { colors } from "../../constants/Colors";
+import { useAppSelector } from "../../hooks/useTypedRedux";
+import LockIcon from "../../assets/icons/LockIcon";
+import { useNavigation } from "@react-navigation/native";
+import { NativeStackNavigationProp } from "@react-navigation/native-stack";
+import { RootStackParamList } from "../../types/Navigation";
+import { screen_names } from "../../constants/ScreenNames";
+
+interface MainHeaderProps {
+	onPress?: () => void; //allow consumer of this component to handle the press event differently in the not-signed-in state.
+}
+
+const MainHeader = (props: MainHeaderProps) => {
+	const { user } = useAppSelector((state) => state.user);
+
+	const navigation =
+		useNavigation<NativeStackNavigationProp<RootStackParamList>>();
+
+	//Temporary/experimental navigation to sign-in/sign-out screen from the profile picture icon
+	const goTOLoginScreen = () => navigation.navigate(screen_names.LOGIN);
```

Comment:
**Nit**: did you intend to name this as `goToLoginScreen`?

Reply:
Yes! Let me correct the typo